command/agent/exec/exec_test.go

```diff
@@ -380,3 +385,164 @@ func TestExecServer_Run(t *testing.T) {
 		})
 	}
 }
+
+func TestExecServer_LogFiles(t *testing.T) {
+	goBinary, err := exec.LookPath("go")
+	if err != nil {
+		t.Fatalf("could not find go binary on path: %s", err)
+	}
+
+	testAppBinary := filepath.Join(os.TempDir(), "test-app")
+
+	if err := exec.Command(goBinary, "build", "-o", testAppBinary, "./test-app").Run(); err != nil {
+		t.Fatalf("could not build the test application: %s", err)
+	}
+	t.Cleanup(func() {
+		if err := os.Remove(testAppBinary); err != nil {
+			t.Fatalf("could not remove %q test application: %s", testAppBinary, err)
+		}
+	})
+
+	tempStderr := filepath.Join(os.TempDir(), "vault-exec-test.stderr.log")
+	t.Cleanup(func() {
+		_ = os.Remove(tempStderr)
+	})
+
+	testCases := map[string]struct {
+		testAppPort   int
+		stderrFile    string
+		expectedError error
+	}{
+		"can_log_to_file": {
+			testAppPort: 34001,
+			stderrFile:  tempStderr,
+		},
+		"cant_open_file": {
+			testAppPort:   34002,
+			stderrFile:    "/file/does/not/exist",
+			expectedError: os.ErrNotExist,
```

Comment:
Why do we error if the file doesn't exist? We open it with `os.O_APPEND|os.O_CREATE|os.O_WRONLY`, so shouldn't be be able to handle writing to a new file?

Reply:
its if the parent folder doesn't exist